ticdc: when start a redo changefeed, ticdc panicked

### What did you do?

1. open a mysql-sink changefeed with redo log

### What did you expect to see?

_No response_

### What did you see instead?

[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
panic: non-positive interval for NewTicker

goroutine 1587 [running]:
time.NewTicker(0xc00025ed60?)
        time/tick.go:22 +0xe5
github.com/pingcap/tiflow/cdc/redo.(*logManager).bgUpdateLog(0xc00cc782a0, {0x7fef9294bb10, 0xc00ce2d660}, 0x0)
        github.com/pingcap/tiflow/cdc/redo/manager.go:497 +0xb8
github.com/pingcap/tiflow/cdc/redo.(*logManager).Run(0xc00cc782a0, {0x7fef9294bb10, 0xc00ce2d660}, {0x4447f00?, 0x462f9c0?, 0x0?})
        github.com/pingcap/tiflow/cdc/redo/manager.go:277 +0x13c
github.com/pingcap/tiflow/cdc/owner.(*changefeed).initialize.func5()
        github.com/pingcap/tiflow/cdc/owner/changefeed.go:596 +0xaa
created by github.com/pingcap/tiflow/cdc/owner.(*changefeed).initialize in goroutine 724
        github.com/pingcap/tiflow/cdc/owner/changefeed.go:594 +0xe99


### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
release-7.5
sh-5.1# /cdc version
Release Version: v7.5.0
Git Commit Hash: 50b7aa338a042cfd79ea2fd969ff9e0fe4464cb5
Git Branch: heads/refs/tags/v7.5.0
UTC Build Time: 2023-11-14 10:44:31
Go Version: go version go1.21.3 linux/amd64
Failpoint Build: false
```

--- a/cdc/redo/manager.go
+++ b/cdc/redo/manager.go
@@ -279,8 +279,18 @@ func (m *logManager) Run(ctx context.Context, _ ...chan<- error) error {
 
 func (m *logManager) getFlushDuration() time.Duration {
 	flushIntervalInMs := m.cfg.FlushIntervalInMs
+	defaultFlushIntervalInMs := redo.DefaultFlushIntervalInMs
 	if m.cfg.LogType == redo.RedoDDLLogFileType {
 		flushIntervalInMs = m.cfg.MetaFlushIntervalInMs
+		defaultFlushIntervalInMs = redo.DefaultMetaFlushIntervalInMs
+	}
+	if flushIntervalInMs < redo.MinFlushIntervalInMs {
+		log.Warn("redo flush interval is too small, use default value",
+			zap.Stringer("namespace", m.cfg.ChangeFeedID),
+			zap.Int("default", defaultFlushIntervalInMs),
+			zap.String("logType", m.cfg.LogType),
+			zap.Int64("interval", flushIntervalInMs))
+		flushIntervalInMs = int64(defaultFlushIntervalInMs)
 	}
 	return time.Duration(flushIntervalInMs) * time.Millisecond
 }

